### ISSUE TITLE
Failed to load module created in test

### ISSUE THREAD

**Issue by Kaciras** [vitest-dev/vitest#6087]
### Describe the bug

After upgrading Vitest to 2.0.2, I get the error when import a file created in test:

```javascript
import { afterEach, expect, it } from "vitest";
import { pathToFileURL } from "url";
import { unlinkSync, writeFileSync } from "fs";

const filename = "bar.js";

afterEach(() => unlinkSync(filename));

it("should works", async () => {
	writeFileSync(filename, "export default 123");
	const mod = await import(pathToFileURL(filename).href);
	expect(mod.default).toBe(123);
});
```

```text
Error: Failed to load url /D:/Coding/JavaScript/vite-svg-sfc/bar.js (resolved id: /D:/Coding/JavaScript/vite-svg-sfc/bar.js). Does the file exist?
    at loadAndTransform (file:///D:/Coding/JavaScript/vite-svg-sfc/node_modules/.pnpm/vite@5.3.3_@types+node@20.14.10/node_modules/vite/dist/node/chunks/dep-CzJTQ5q7.js:52400:17)
    at async ViteNodeServer._transformRequest (file:///D:/Coding/JavaScript/vite-svg-sfc/node_modules/.pnpm/vite-node@2.0.2_@types+node@20.14.10/node_modules/vite-node/dist/server.mjs:524:16)
    at async ViteNodeServer._fetchModule (file:///D:/Coding/JavaScript/vite-svg-sfc/node_modules/.pnpm/vite-node@2.0.2_@types+node@20.14.10/node_modules/vite-node/dist/server.mjs:486:17)
    at async Proxy.fetch (file:///D:/Coding/JavaScript/vite-svg-sfc/node_modules/.pnpm/vitest@2.0.2_@types+node@20.14.10/node_modules/vitest/dist/vendor/cli-api.FtQZKvQk.js:8093:22)
    at async EventEmitter.<anonymous> (file:///D:/Coding/JavaScript/vite-svg-sfc/node_modules/.pnpm/vitest@2.0.2_@types+node@20.14.10/node_modules/vitest/dist/vendor/index.BpSiYbpB.js:69:20)
```

It works in Vitest 1.6.0

### Reproduction

See above

### System Info

```shell
System:
    OS: Windows 11 10.0.22621
    CPU: (12) x64 AMD Ryzen 5 5625U with Radeon Graphics
    Memory: 6.20 GB / 15.31 GB
  Binaries:
    Node: 22.4.1 - D:\Program Files\NodeJS\node.EXE
    npm: 10.8.1 - D:\Program Files\NodeJS\npm.CMD
    pnpm: 9.5.0 - ~\AppData\Roaming\npm\pnpm.CMD
    bun: 1.1.18 - ~\.bun\bin\bun.EXE
  Browsers:
    Edge: Chromium (124.0.2478.67)
    Internet Explorer: 11.0.22621.3527
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by AriPerkkio** [comment]
Works fine for me: 

<img src="https://github.com/vitest-dev/vitest/assets/14806298/16ea26d9-b41e-48d6-9d3b-78a09f0ba12e" height="200" />

**Comment by Kaciras** [comment]
Seems to be an error only on Windows.

https://github.com/Kaciras/vitest-issue-1/actions/runs/9886516528/job/27306432893

**Comment by meteorlxy** [comment]
Same error in vuepress. Only failed on windows.

https://github.com/vuepress/core/actions/runs/9934091779/job/27437900959

**Comment by syi0808** [comment]
> Same error in vuepress. Only failed on windows.
> 
> https://github.com/vuepress/core/actions/runs/9934091779/job/27437900959

I think this is fixed by #6109. Can you wait for the release?
